### DISCUSSION TITLE
Slider: how to toggle disabled state?

### DISCUSSION THREAD

**Discussion by nephix** [chakra-ui/zag#998] [Q&A] [answered]
Hey zaggers 👋 

How can I toggle the disabled state on a [slider](https://zagjs.com/components/react/slider) component?

I have a minimal example [here](https://codesandbox.io/s/intelligent-turing-9qrvf8?file=/src/Slider.js) where I simply pass the `disabled` prop to `useMachine`.

When clicking the `toggle disabled` button you can see that the thumb opacity changes, but you are still able to change the slider value. (Thumb doesn't move but I think that's just to due my minimal example)

Ideally I would like to:

- change `disabled` state via props, or
- have an `api.setDisabled()` to set it from within the component

Any ideas how that can be achieved?

**Comment by segunadebayo** [comment, marked as answer]
In this case, you can update the machine context transiently. 

![CleanShot 2023-11-08 at 12 48 51@2x](https://github.com/chakra-ui/zag/assets/6916170/1f930d0e-8ef9-467f-a276-985a12fd6075)

Here's the Sandbox:
https://codesandbox.io/s/musing-banach-k882pd?file=/src/Slider.js

**Reply by nephix** [reply]
Ah I didn't know, thanks a lot @segunadebayo 🙏
